src/httpProblemDetails.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpProblemDetailsDefinition } from './httpProblemDetailsDefinition';
+
+const invalidDetailName = ['type', 'status', 'title'];
+
+export class HttpProblemDetails {
+  public readonly type: string;
+  public readonly title: string;
+  public readonly status: number;
+  private additionalDetails: Map<string, any>;
+
+  constructor({ type, title, status }: HttpProblemDetailsDefinition) {
+    this.type = type;
+    this.title = title;
+    this.status = status;
+    this.additionalDetails = new Map<string, any>();
+  }
+
+  public addAdditionalDetail(key: string, value: any) {
+    if (invalidDetailName.includes(key)) {
+      throw new Error(`${key} is forbidden details name`);
+    }
+    if (value) {
+      this.additionalDetails.set(key, value);
+    }
+  }
+
+  toJson(): Record<string, any> {
+    const additionalDetails: Record<string, any> = {};
```

Comment:
What's the difference between `Map<string, any>` and `Record<string, any>`? Why do you need one over the other?

Reply:
I mean, maybe `private additionalDetails` should be of type `Record<string, any>` in the first place, and then you don't need this mapping here?

Reply:
> What's the difference between Map<string, any> and Record<string, any>? Why do you need one over the other?
So basically you are using `Map` and on the second option you are using just simple `Object`. Hmm, I think that I don't see any reason to use Map instead of just a simple object 🤔